src/Microsoft.ML.Data/Evaluators/Metrics/BinaryPrecisionRecallDataPoint.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// This class represents one data point on Precision-Recall curve for binary classification.
+    /// </summary>
+    public sealed class BinaryPrecisionRecallDataPoint
+    {
+        /// <summary>
+        /// Gets the threshold for this data point.
+        /// </summary>
+        public double Threshold { get; }
+        /// <summary>
+        /// Gets the precision for the current threshold.
+        /// </summary>
+        public double Precision { get; }
+        /// <summary>
+        /// Gets the recall for the current threshold.
+        /// </summary>
+        public double Recall { get; }
+
+        /// <summary>
+        /// Gets the true positive rate for the current threshold.
+        /// </summary>
+        public double TruePositiveRate => Recall;
+
+        /// <summary>
+        /// Gets the false positive rate for the given threshold.
+        /// </summary>
+        public double FalsePositiveRate { get; }
+
+        internal BinaryPrecisionRecallDataPoint(ValueGetter<float> thresholdGetter, ValueGetter<double> precisionGetter, ValueGetter<double> recallGetter, ValueGetter<double> fprGetter)
```

Comment:
>ValueGetter<double> precisionGetter [](start = 84, length = 35)

I would recommend that all logic persuant to the getters remain in the same block of code as the cursor (anything else is just a pointless increase of complexity), but this at least avoids the main problem so, I guess we can fix it later.

Reply:
sure, we can do it in this or in a separate PR.

---
In reply to: [268370725](https://github.com/dotnet/machinelearning/pull/3039#discussion_r268370725) [](ancestors = 268370725)